pkgs/by-name/sm/smartcat/package.nix

```
@@ -0,0 +1,42 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+
+, openssl
+, pkg-config
+, xorg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "smartcat";
+  version = "1.3.0";
+  src = fetchFromGitHub {
+    owner = "efugier";
+    repo = "smartcat";
+    rev = "main";
+    hash = "sha256-iCtNNKXo0peGGUaQXKaaYaEo7MAL70PX0BAWPERNmlo=";
+  };
+
+  cargoLock.lockFile = "${src}/Cargo.lock";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl.dev
+    xorg.libX11.dev
+  ];
+
+  PKG_CONFIG_PATH = lib.makeSearchPath "lib/pkgconfig" buildInputs;
+
+  meta = with lib; {
+    description = "A cat-like command to integrate large language models into the command line";
```

```suggestion
    description = "Integrate large language models into the command line";
```

per https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes, should not start with "a" - also, it arguably has nothing to do with cat

---

Right, fair enough. I was trying to somewhat capture the original description's vibes but I think I overcompensated so that's for the best.